app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -0,0 +1,116 @@
+module MiqServer::WorkerManagement::Monitor::Kubernetes
+  extend ActiveSupport::Concern
+  attr_accessor :pod_resource_version
+
+  def current_pods
+    @current_pods ||= Concurrent::Hash.new
+  end
+
+  def reset_current_pods
+    current_pods.clear
+  end
+
+  def cleanup_failed_deployments
+    ensure_pod_monitor_started
+    delete_failed_deployments
+  end
+
+  def failed_deployments(restart_count = 5)
+    # TODO: This logic might flag deployments that are hitting memory/cpu limits or otherwise not really 'failed'
+    current_pods.select { |_name, h| h[:last_state_terminated] && h.fetch(:container_restarts, 0) > restart_count }.collect { |_name, h| h[:label_name] }
+  end
+
+  private
+
+  def start_pod_monitor
+    @monitor_thread ||= begin
+      Thread.new { monitor_pods }
+    end
```

Comment:
```suggestion
    @monitor_thread ||= Thread.new { monitor_pods }
```

begin/end is not needed

Reply:
yeah, thanks... I removed the other lines in the body of the begin block and don't need it anymore.